### PR TITLE
CFM-1206 Implement CTA hook for group actions

### PR DESCRIPTION
**Pull request by DavidHernandez**
Issues: https://issuetracker.amplexor.com/jira/browse/CFM-1206

Adds the next CTAs:

Group member
    GM and draft content in this group: X unpublished items
Group admin
    Manage this group
    There are X new membership requests

The rest have been added in others PRs/Issues

Also, brings changes from develop.

### CODE REVIEW

**Comment by ordavidil** [comment]
@DavidHernandez 
It seems that this PR is not compared to the right branch ...

**Comment by DavidHernandez** [comment]
@ordavidil it is compared with the right branch, but I merged the changes from develop, so a lot of extra commits appear on this branch. Should I merge develop into the parent branch too so those commits only appear on the parent branch?

**Comment by ordavidil** [comment]
of course ...

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_og/c4m_user_og.module [hunk 1]
```diff
@@ -471,36 +471,62 @@ function c4m_user_og_c4m_cta_block() {
     return;
   }
 
-  if (!user_is_anonymous()) {
+  $items = array();
+  if (user_is_anonymous()) {
+    $types = array(
+      'open' => array(
+        'button_label' => 'Join this group',
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_og/c4m_user_og.module]: `t()`

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_og/c4m_user_og.module [hunk 2]
```diff
@@ -471,36 +471,62 @@ function c4m_user_og_c4m_cta_block() {
...
+        'button_icon' => 'fa-user-plus',
+      ),
+      'moderated' => array(
+        'button_label' => 'Request membership for this group',
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_og/c4m_user_og.module]: `t()`

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_og/c4m_user_og.module [hunk 3]
```diff
@@ -471,36 +471,62 @@ function c4m_user_og_c4m_cta_block() {
...
+  else {
     global $user;
 
     if (og_get_membership('node', $group->nid, 'user', $user->uid)) {
-      // Current user is already a member of this group.
-      return;
+      if (c4m_og_user_is_owner('node', $group->nid)) {
+        $items['manage'] = array(
+          '#type' => 'markup',
+          '#markup' => l('Manage this group', '/manage'),
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_og/c4m_user_og.module]: `t()`

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_og/c4m_user_og.module [hunk 4]
```diff
@@ -471,36 +471,62 @@ function c4m_user_og_c4m_cta_block() {
...
+        if ($pending_requests = c4m_og_get_all_group_memberships_by_state($group->nid, OG_STATE_PENDING)) {
+          $items['memberships'] = array(
+            '#type' => 'markup',
+            '#markup' => t('There are %requests new membership requests.', array('%requests' => $pending_requests)),
+          );
+        }
+      }
+      else {
+        $variables = array(
+          'bundles' => array('wiki_page', 'photoalbum', 'discussion', 'event', 'document', ''),
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_og/c4m_user_og.module]: @DavidHernandez 
why do we need an empty element over here ?

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_og/c4m_user_og.module [hunk 5]
```diff
@@ -471,36 +471,62 @@ function c4m_user_og_c4m_cta_block() {
...
+          'state' => 0,
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_og/c4m_user_og.module]: Use a constant instead of `0`

Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_og/c4m_user_og.module]: probably `NODE_NOT_PUBLISHED`

**Comment by ordavidil** [comment]
@DavidHernandez 
Can you please add comments next to each condition to explain the case ?
thanks ...

**Comment by jover** [comment]
Feedback on ticket CFM-1206

**Pull request closed without merging**